ext/openssl/ossl_ssl.c

```diff
@@ -1570,13 +1594,16 @@ ossl_start_ssl(VALUE self, int (*func)(), const char *funcname, VALUE opts)
 		if (!verify_msg)
 		    verify_msg = "(null)";
 		ossl_clear_error(); /* let ossl_raise() not append message */
-		ossl_raise(eSSLError, "%s returned=%d errno=%d state=%s: %s (%s)",
-			   funcname, ret2, errno, SSL_state_string_long(ssl),
+		ossl_raise(eSSLError, "%s returned=%d errno=%d peeraddr=%s state=%s: %s (%s)",
```

Comment:
This looks good to me. I wonder if we should expose it as an attribute on the error too, i.e. `error.peer_address` or something similar.

Reply:
I think that makes sense, it'd be useful to be able to access this information. Are you thinking something like this (Ruby equivalent)? Not sure what exactly this would look like in C.
```ruby
class SSLError < ...
  attr_reader :peer_sockaddr

  def initialize(message, peer_sockaddr)
    @message = message
    @peer_sockaddr = peer_sockaddr
  end
end

# then when raising
raise SSLError.new(message, peer_sockaddr)
```

Reply:
I'm not sure about the usefulness because the current approach isn't a reliable way to get the peer's address. The underlying socket must still be alive. It probably is the case and it is useful to have in exception message, but that would be unusable for complete diagnosis, which I think is the intended use case of such an attribute.

We need to store the peer address before we attempt the handshake to report it reliably. This is an overhead and I honestly don't think it's worth. The user can store the connecting address to a safe place themselves.